lib/inertia_rails/renderer.rb

```diff
@@ -112,5 +115,32 @@ def resolve_component(component)
 
       configuration.component_path_resolver(path: controller.controller_path, action: controller.action_name)
     end
+
+    def validate_partial_props(merged_props)
+      return unless rendering_partial_component?
+
+      unrequested_props = merged_props.select do |key, prop|
```

Comment:
Is this based on what the frontend requests?

Reply:
Yea, if the frontend requests only `:search` as part of a partial reload, `unrequested_props` will have every prop _except for_ `:search`.

Then below we filter out any props that aren't callable, leaving us with props that were defined as values.